src/schema.rs

```diff
@@ -65,25 +60,33 @@ impl Schema {
 
 /// Provides an Iterator which returns [Type]s inside a [Schema]
 pub struct SchemaTypeIterator {
-    types: Rc<TypeStore>,
+    type_store: Rc<TypeStore>,
     index: usize,
+    types: Vec<TypeId>,
```

Comment:
Allocating a `Vec` for each iterator strikes me as a bit expensive. Now that this iterator holds a reference to the type store, could we access the types via the `Rc<TypeStore>`?